### DISCUSSION TITLE
How to complete the PR checklist

### DISCUSSION THREAD

**Discussion by callumrollo** [GliderToolsCommunity/GliderTools#111] [Show and tell]
When you create a PR, you will be prompted to complete a checklist. Each line of the checklist will render as a checkbox on github. Put an x in the square brackets to make a check like so:

`- [x] a complete item`

This will render as:

- [x] a complete item

**Item by item explanation**:

 - [ ] Closes #xxxx
 
 Closes Issue number #xxxx. This will link that issue and automatically close it when the PR is accepted. Most PRs should have an associated Issue. e.g. [PR 105](https://github.com/GliderToolsCommunity/GliderTools/pull/105) closed [Issue 16](https://github.com/GliderToolsCommunity/GliderTools/issues/16). The # is what that makes this happen automagically.
 
 - [ ] Tests added
 
 We should include tests when adding new code. If an adequate test already exists for the function you are changing, you don't need to add a new one
 
 - [ ] Passes `pre-commit run --all-files`
 
 The pre-commit includes stuff like linting. Running it on your machine before commiting will prevent you failing the code-style check
 
 - [ ] User visible changes (including notable bug fixes) are documented in `whats-new.rst`
 
 If you are adding new functionality, detail it over at [whats-new.rst](https://github.com/GliderToolsCommunity/GliderTools/blob/master/docs/whats-new.rst) so users will be aware of it.
 
 - [ ] New functions/methods are listed in `api.rst`
 
 As above in [api.rst](https://github.com/GliderToolsCommunity/GliderTools/blob/master/docs/api.rst) this adds your function to the map of the package
 
If any of the checks do not apply to your code, you can delete them when making your PR.

**Comment by isgiddy** [comment]
Thanks @callumrollo 

Adding here some detail on pre-commit that you shared that was helpful for me :) 

After committing new changes on a branch for the first time: 

```
pip install pre-commit
pre-commit run --all-files
```

If you get a fail, you should see your file has been changed (red/green colours).
Do `pre-commit run --all-files` again and it should pass.    
You then do another commit with message "linting" or similar and push. The CI will run again on your PR and you should pass the PR.
